csrc/device_lower/pass/insert_syncs.cpp

```diff
@@ -20,6 +20,40 @@ namespace nvfuser {
 
 namespace {
 
+// Determine if any for loop is a LoadWarp circular buffering stage
+bool isWithinLoadWarp(const std::vector<ForLoop*> for_loops) {
+  return std::any_of(for_loops.begin(), for_loops.end(), [](ForLoop* fl) {
+    return fl->circularBufferLoopStage() == CircularBufferLoopStage::LoadWarp;
+  });
+}
+
+// Determine if any for loop is a ComputeWarp circular buffering stage
+bool isWithinComputeWarp(const std::vector<ForLoop*> for_loops) {
+  return std::any_of(for_loops.begin(), for_loops.end(), [](ForLoop* fl) {
+    return fl->circularBufferLoopStage() ==
+        CircularBufferLoopStage::ComputeWarp;
+  });
+}
+
+// Return true if any for loop is ComputeWarp.
+// Return false if any for loop is LoadWarp.
+// Return std:nullopt if none of the for loops are a warp specialized stage.
+std::optional<bool> isOptionalLoadOrComputeSync(
```

Comment:
Is `isOptionalComputeSync?` a better name?

Reply:
I thought it was important to indicate optional, load, or compute in the name for clarity, but it is kinda long.